### PR TITLE
tests: don't make noise

### LINKED ISSUE #2548: Some UnitTests make noise
Ha! While running `UnitTests.testAll` I started hearing some strange noises coming from my computer -- looks like the tests are really running :)

The sounds are reproduceable and play while these lines are being printed:

> PASS:a TestCoreUGens:test_muladd - "ar5.madd(ar5, kr2)"                                                                                
> PASS:a TestCoreUGens:test_bufugens - data->loadCollection->PlayBuf->RecordBuf->loadToFloatArray->data (1 channels)                     
> PASS:a TestCoreUGens:test_bufugens - data->loadCollection->PlayBuf->RecordBuf->loadToFloatArray->data (2 channels)                     
> PASS:a TestCoreUGens:test_bufugens - data->loadCollection->PlayBuf->RecordBuf->loadToFloatArray->data (8 channels)                     
> PASS:a TestCoreUGens:test_bufugens - data->loadCollection->PlayBuf->RecordBuf->loadToFloatArray->data (16 channels)                    
> PASS:a TestCoreUGens:test_bufugens - data->loadCollection->PlayBuf->RecordBuf->loadToFloatArray->data (32 channels)                    
> PASS:a TestCoreUGens:test_bufugens - data->loadCollection->PlayBuf->RecordBuf->loadToFloatArray->data (33 channels)                    
> PASS:a TestCoreUGens:test_demand - Duty should free itself after a limited sequence                                                    
> PASS:a TestCoreUGens:test_demand - Duty+LFTri should not output NaN                                                                    
> PASS:a TestCoreUGens:test_pitchtrackers - ZCR.ar() tracking a SinOsc

### PR DESCRIPTION
**Pull request by snappizz**
This commit silences several unit tests that make noise. Fix #2548.

### CODE REVIEW

File: testsuite/classlibrary/TestCoreUGens.sc [hunk 1]
```diff
@@ -341,7 +341,8 @@ test_bufugens{
 
 		// Copying data from b to c:
 		{
-			RecordBuf.ar(PlayBuf.ar(numchans, b, BufRateScale.ir(b), doneAction: 2), c, loop:0) * 0.1;
+			RecordBuf.ar(PlayBuf.ar(numchans, b, BufRateScale.ir(b), doneAction: 2), c, loop:0);
+            DC.ar(0);
```
Comment (telephon) [inline testsuite/classlibrary/TestCoreUGens.sc]: I's also OK to return `0` from a function that you play. You can also output a control rate signal. (The relevant source code is in `GraphBuilder`).

File: testsuite/classlibrary/TestCoreUGens.sc [hunk 2]
```diff
@@ -373,7 +374,7 @@ test_demand {
 	o.add;
 
 	tests = [
-		{LPF.ar(LeakDC.ar(Duty.ar(0.1, 0, Dseq((1..8)), 2)))}
+            {LPF.ar(LeakDC.ar(Duty.ar(0.1, 0, Dseq((1..8)), 2))); DC.ar(0);}
```
Comment (telephon) [inline testsuite/classlibrary/TestCoreUGens.sc]: same here …

File: testsuite/classlibrary/TestCoreUGens.sc [hunk 3]
```diff
@@ -393,7 +394,7 @@ test_demand {
 			5453, {testNaN = msg[3] <= 0.0 or:{msg[3] >= 1.0}}		);
 	});
 	o.add;
-	{Line.kr(1, 0, 1, 1, 0, 2); SendTrig.kr(Impulse.kr(10, 0.5), 5453, LFTri.ar(Duty.ar(0.1, 0, Dseq(#[100], 1))))}.play;
+    {Line.kr(1, 0, 1, 1, 0, 2); SendTrig.kr(Impulse.kr(10, 0.5), 5453, LFTri.ar(Duty.ar(0.1, 0, Dseq(#[100], 1)))); DC.ar(0)}.play;
```
Comment (telephon) [inline testsuite/classlibrary/TestCoreUGens.sc]: also here.

**Comment by snappizz** [comment]
Thanks Julian, it's all taken care of.

**Comment by telephon** [comment]
OK, good! Of course I also hear @danstowell 's voice saying that it is nice, we shouldn't get rid of it?

**Comment by crucialfelix** [comment]
My voice too. I think it's hilarious and awesome to hear the unit tests
sputtering away. Crucial library tests make plenty of noise

On Sat, Dec 10, 2016, 19:40 Julian Rohrhuber <notifications@github.com>
wrote:

OK, good! Of course I also hear @danstowell <https://github.com/danstowell>
's voice saying that it is nice, we shouldn't get rid of it?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/supercollider/supercollider/pull/2549#issuecomment-266232527>,
or mute the thread

**Comment by vivid-synth** [comment]
Haha cool, as long as it's expected behavior I'm fine with keeping it :)

**Comment by snappizz** [comment]
No objections from me.

**Pull request closed without merging**